src/hooks/useAnnouncer/useAnnouncer.ts

```diff
@@ -0,0 +1,69 @@
+import { useRef } from "react";
+
+/* Due to possibility of multiple Carbon versions being used on the page,
+ * it is neccessary to maintain the same values for the following ids */
+const POLITE_REGION_ID = "__CARBON_INTERVALS_POLITE_LIVE_REGION";
+const ASSERTIVE_REGION_ID = "__CARBON_INTERVALS_ASSERTIVE_LIVE_REGION";
+
+const TIMEOUT_DURATION = 200;
+type Assertiveness = "polite" | "assertive";
+
+function getRegion(assertiveness: Assertiveness): HTMLElement {
+  const id =
+    assertiveness === "assertive" ? ASSERTIVE_REGION_ID : POLITE_REGION_ID;
+  const region = document.getElementById(id);
+
+  if (region) return region;
+
+  const element: HTMLDivElement = document.createElement("div");
+  const visuallyHiddenStyles: Partial<CSSStyleDeclaration> = {
+    clip: "rect(0 0 0 0)",
+    clipPath: "inset(50%)",
+    height: "1px",
+    overflow: "hidden",
+    position: "absolute",
+    whiteSpace: "nowrap",
+    width: "1px",
+  };
+  element.setAttribute("id", id);
+  element.setAttribute(
+    "role",
+    assertiveness === "assertive" ? "alert" : "status"
+  );
+  element.setAttribute("aria-live", assertiveness);
+  element.setAttribute("aria-atomic", "false");
+  element.setAttribute("aria-relevant", "additions");
+  Object.assign(element.style, visuallyHiddenStyles);
+
+  return document.body.appendChild(element);
+}
+
+function useAnnouncer(): {
+  announcePolitely: (message: string) => void;
+  announceAssertively: (message: string) => void;
+} {
+  const politeRegion = useRef<HTMLElement>(getRegion("polite"));
+  const assertiveRegion = useRef<HTMLElement>(getRegion("assertive"));
+
+  // use .textContent instead of .innerText, which can't be used to get text of elements hidden with css.
+  function announcePolitely(message: string): void {
+    politeRegion.current.textContent = message;
+    setTimeout(() => {
+      politeRegion.current.textContent = "";
+    }, TIMEOUT_DURATION);
```

Comment:
My intention is to change the design of announcement functions to append `<p>` elements into the live region with the dispatched message inside, as opposed to setting the text content, so we are avoid having a race condition where multiple messages from different micro-frontends are incoming at a similar time.

Reply:
I think @robinzigmond is right about needing to maintain a global queue here, we'll need to make sure that the timeouts are only triggered when the next message is popped off etc